### DISCUSSION TITLE
umi@3.5.20中支持配置webpack中的cache.cacheDirectory吗？

### DISCUSSION THREAD

**Discussion by lxow456** [umijs/umi#12122] [Q&A] [answered]
&emsp;为了使用webpack 5的cache特性，我开启了`webpack5: {}`。但是我看`webpack5`在文档里没有提到支持webpack cache的其他配置项。比如`cache.cacheDirectory`。   

![image](https://github.com/umijs/umi/assets/50816861/52509cf0-3c25-414f-9849-3f5ea22962d8)

&emsp;我看默认的cacheDirectory是在`.umi-production\.cache\webpack`里，有办法能改成`node_modules\.cache\webpack`吗？   
&emsp;我的umi版本是3.5.20

**Comment by fz6m** [comment, marked as answer]
umi 3 不维护了，建议升级到 umi 4 。

配这个选项对大部分开发的场景没什么用，持久化缓存只对第二次构建才生效，一般这个持久化缓存多用于在云构建环境里由基础设施缓存上一次的 webpack 缓存，并在下一次构建开始时自动恢复缓存到指定位置，并精确的指定失效的依赖条件有哪些，通常情况下用不到。
